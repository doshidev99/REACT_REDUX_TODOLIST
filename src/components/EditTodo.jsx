import React, { useState } from 'react';

export default function EditTodo({ completed, text, onSave }) {
  const [value, setValue] = useState(text);
  const [isEdit, setEdit] = useState(false);
  function handleSubmit(e) {
    const text = e.target.value.trim();
    if (e.which === 13) {
      setEdit(false);
      onSave(text);
    }
  }
  function handleValueChange(e) {
    setValue(e.target.value);
  }
  const handleSubmitBlur = e => {
    setEdit(false);
    onSave(text);
  };

  return (
    <div>
      {!isEdit ? (
        <p
          style={{
            textDecoration: completed ? 'line-through' : 'none',
          }}
          onDoubleClick={() => setEdit(true)}
        >
          {value}
        </p>
      ) : (
        <input
          type="text"
          autoFocus
          value={value}
          onKeyDown={handleSubmit}
          onBlur={e => handleSubmitBlur(e)}
          onChange={handleValueChange}
        />
      )}
    </div>
  );
}
