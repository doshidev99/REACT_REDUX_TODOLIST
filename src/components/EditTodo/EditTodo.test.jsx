import React from 'react';
import { shallow } from 'enzyme';
import EditTodo from './index';

describe('EditTodo', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<EditTodo debug />);

    expect(component).toMatchSnapshot();
  });
});
