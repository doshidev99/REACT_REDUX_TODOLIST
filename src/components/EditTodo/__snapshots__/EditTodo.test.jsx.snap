// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTodo should render correctly in "debug" mode 1`] = `
<div>
  <p
    onDoubleClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  />
</div>
`;
